src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs

```diff
@@ -2,75 +2,67 @@
 // Licensed under the MIT License.
 
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
 using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureAnalyzerVerifier<Azure.ClientSdk.Analyzers.ClientAssemblyNamespaceAnalyzer>;
 
 namespace Azure.ClientSdk.Analyzers.Tests
 {
     public class AZC0001Tests
     {
-        private readonly DiagnosticAnalyzerRunner _runner = new DiagnosticAnalyzerRunner(new ClientAssemblyNamespaceAnalyzer());
-
         [Fact]
         public async Task AZC0001ProducedForInvalidNamespaces()
         {
-            var testSource = TestSource.Read(@"
-namespace /*MM*/RandomNamespace
+            const string code = @"
+namespace RandomNamespace
 {
     public class Program { }
-}
-");
-            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
-            var diagnostic = Assert.Single(diagnostics);
-            Assert.Equal("AZC0001", diagnostic.Id);
-            Assert.Equal("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
-                         "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure", diagnostic.GetMessage());
+}";
 
-            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+            var diagnostic = Verifier.Diagnostic("AZC0001")
+                .WithMessage("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
+                             "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure")
+                .WithSpan(2, 11, 2, 26);
+
+            await Verifier.VerifyAnalyzerAsync(code, diagnostic);
         }
 
         [Fact]
-        public async Task AZC0001ProducedOneErrorPerNamspaceDefinition()
+        public async Task AZC0001ProducedOneErrorPerNamespaceDefinition()
         {
-            var testSource = TestSource.Read(@"
-namespace RandomNamespace
+            const string code = @"
+namespace [|RandomNamespace|]
```

Comment:
What is `[|` for?

Reply:
This marks the place where diagnostic would be.